test/unit/dimensions.js

```diff
@@ -694,13 +696,6 @@ QUnit.test( "interaction with scrollbars (gh-3589)", function( assert ) {
 			borderBox.clone().css( { overflow: "auto" } ).appendTo( parent )[ 0 ].offsetWidth -
 			borderBox[ 0 ].offsetWidth;
 
-	if ( borderBoxLoss > 0 ) {
-		borderBox.css( {
-			width: ( size + borderBoxLoss ) + "px",
-			height: ( size + borderBoxLoss ) + "px"
-		} );
-	}
```

Comment:
It took some self-convincing, but this change seems to be OK. However, I do like the original style better because `borderBox.innerWidth(99); … assert.equal(borderBox.innerWidth(), 99)` is more direct than `borderBox.innerWidth(82); … assert.equal(borderBox.innerWidth() + 17, 99)`, especially in comparison with the corresponding `contentBox` assertions (whose values are always 6 less, to account for 1px border width and 2px of padding on each edge).
<details><summary><code>borderBox</code> evaluation diff</summary>

```diff
--- before
+++ after
 size = 100
 borderBoxLoss = 17
-borderBox.css( { width: "117px", height: "117px" } )
 i = 0
 assert.equal( contentBox.innerWidth/innerHeight(), 104 )
 assert.equal( contentBox.outerWidth/outerHeight(), 106 )
-assert.equal( borderBox.innerWidth/innerHeight(), 98 )
-assert.equal( borderBox.outerWidth/outerHeight(), 100 )
+assert.equal( borderBox.innerWidth/innerHeight() + 17, 98 )
+assert.equal( borderBox.outerWidth/outerHeight() + 17, 100 )
 i = 1
 size = 101
-borderBox.innerWidth( 98 + 1 ).innerHeight( 98 + 1 )
+borderBox.innerWidth( 81 + 1 ).innerHeight( 81 + 1 )
 assert.equal( contentBox.innerWidth/innerHeight(), 105 )
 assert.equal( contentBox.outerWidth/outerHeight(), 107 )
-assert.equal( borderBox.innerWidth/innerHeight(), 99 )
-assert.equal( borderBox.outerWidth/outerHeight(), 101 )
+assert.equal( borderBox.innerWidth/innerHeight() + 17, 99 )
+assert.equal( borderBox.outerWidth/outerHeight() + 17, 101 )
 i = 2
 size = 103
-borderBox.outerWidth( 101 + 2 ).outerHeight( 101 + 2 )
+borderBox.outerWidth( 84 + 2 ).outerHeight( 84 + 2 )
 assert.equal( contentBox.innerWidth/innerHeight(), 107 )
 assert.equal( contentBox.outerWidth/outerHeight(), 109 )
-assert.equal( borderBox.innerWidth/innerHeight(), 101 )
-assert.equal( borderBox.outerWidth/outerHeight(), 103 )
+assert.equal( borderBox.innerWidth/innerHeight() + 17, 101 )
+assert.equal( borderBox.outerWidth/outerHeight() + 17, 103 )
```
</details>

Reply:
Update: No, these changes are in fact wrong. And we need that initial `.css(…)` call to reveal the bug, because the difference is a constant offset after that.

Reply:
Sorry what bug does it reveal exactly? The element is still being shrunk, but we're making it shrink to the the value we expect in the assertion instead of changing the assertion. I don't really see a logical difference. Either way, we're accounting for the size difference in IE9. I don't really mind your code suggestion. I'm just not seeing any advantage other than a cosmetic one.

Reply:
I went ahead and changed it, but I'd still like to understand how the previous was wrong.

Reply:
I think it's because of the particularly strange manifestation of #3699... we want to have a dimension set before the assertions. Things might still work otherwise, but I definitely saw some passes that should have been failures when I was experimenting. And if I remember correctly from the original ticket, there were scenarios where the reported values didn't agree with the painted dimensions.

Bottom line: "wrong" is probably too strong, but it is more reliable to ensure a set before testing against expectations.